src/features/wallet/send/SendTransferForm.tsx

```diff
@@ -0,0 +1,420 @@
+        setAddress(ensAddress)
+        setAddressLoading(false)
+        return
+      }
+    }
+    setAddressLoading(false)
+    setAddressAlias('')
+  }, [address])
+
+  // Update the internal HNT amount based on form input
+  useEffect(() => {
+    const hntBalance = Balance.fromFloat(
+      parseFloat(amount),
+      CurrencyType.networkToken,
+    )
+    setBalanceAmount(hntBalance)
+  }, [amount])
+
+  // Update the required fee if HNT amount changes
+  useAsync(async () => {
+    updateFee()
+  }, [balanceAmount, transferData?.amountToSeller])
+
+  const getNonce = (): number => {
+    if (!account?.speculativeNonce) return 1
+    return account.speculativeNonce + 1
+  }
+
+  const updateFee = async () => {
+    let dcFee
+    if (type === 'payment') {
+      dcFee = await calculatePaymentTxnFee(
+        balanceAmount.integerBalance,
+        getNonce(),
+        address,
+      )
+    } else if (type === 'dc_burn') {
+      dcFee = await calculateBurnTxnFee(
+        balanceAmount.integerBalance,
+        address,
+        getNonce(),
+        memo,
+      )
+    } else if (type === 'transfer') {
+      dcFee = await calculateTransferTxnFee(transferData?.partialTransaction)
+    } else {
+      throw new Error('Unsupported transaction type')
+    }
+    const hntFee = feeToHNT(dcFee)
+    setFee(hntFee)
+  }
+
+  // Helper to normalize direct "amount" input value
+  const setFormAmount = (formAmount: string) => {
+    if (formAmount === decimalSeparator || formAmount.includes('NaN')) {
+      return setAmount(`0${decimalSeparator}`)
+    }
+    const rawInteger = (formAmount.split(decimalSeparator)[0] || formAmount)
+      .split(groupSeparator)
+      .join('')
+    const integer = parseInt(rawInteger, 10).toLocaleString(locale)
+    let decimal = formAmount.split(decimalSeparator)[1]
+    if (integer === 'NaN') {
+      return setAmount('')
+    }
+    if (decimal && decimal.length >= 9) decimal = decimal.slice(0, 8)
+    setAmount(
+      formAmount.includes(decimalSeparator)
+        ? `${integer}${decimalSeparator}${decimal}`
+        : integer,
+    )
+  }
+
+  const setMaxAmount = () => {
+    triggerNavHaptic()
+    const balance = account?.balance
+    if (!balance) return
+    if (fee > balance) {
+      const balanceStr = balance.toString(8, {
+        decimalSeparator,
+        groupSeparator,
+        showTicker: false,
+      })
+      setAmount(balanceStr)
+    } else {
+      const maxAmount = balance.minus(fee)
+      const maxAmountStr = maxAmount.toString(8, {
+        decimalSeparator,
+        groupSeparator,
+        showTicker: false,
+      })
+      setAmount(maxAmountStr)
+    }
+  }
+
+  const renderLockedPaymentForm = () => [
+    <LockedField label={t('send.address.label')} value={address} />,
+    <LockedField
+      label={t('send.amount.label')}
+      value={amount}
+      footer={amount ? <FeeFooter fee={fee} /> : undefined}
+      bottom
+    />,
+  ]
+
+  const renderLockedBurnForm = () => [
```

Comment:
This shouldn't return an array, it should use `(` and `)` instead of `[` and `]`